src/Components/Components/src/RenderTree/Renderer.cs

```diff
@@ -582,9 +583,10 @@ private ComponentState GetRequiredRootComponentState(int componentId)
         /// </summary>
         protected virtual void ProcessPendingRender()
         {
-            if (Disposed)
+            if (_rendererIsDisposed)
             {
-                throw new ObjectDisposedException(nameof(Renderer), "Cannot process pending renders after the renderer has been disposed.");
+                // Once we're disposed, we'll disregard further attempts to render anything
+                return;
```

Comment:
Previously we didn't get this far, because an earlier `Dispatcher.CheckAccess` call would have thrown. But now we do get this far so we need to not regard it as an error any more. It's now just a no-op to try rendering after disposal.